01-JS-DOM-APIs/script.js

```diff
@@ -0,0 +1,120 @@
+// exercise 6
+function show(){
+    let obj = document.getElementById("hidden");
+    obj.style.visibility ="visible";
+}
+
+function DoAJoke() {
+  let xhttp = new XMLHttpRequest();
+
+  xhttp.open("GET", "http://api.icndb.com/jokes/random", true);
+
+  xhttp.onreadystatechange = function(event) {
+        if (xhttp.readyState === XMLHttpRequest.DONE) {
+          let response = JSON.parse(event.target.response);
+          document.getElementById("hidden").innerHTML = response.value.joke;
+        }
+   }
+
+   xhttp.send();
+}
+
+
+// exercise 7
+function secondJoke(){
+ let config = {
+   url: "http://api.icndb.com/jokes/random"
+ }
+
+ let xhttp = new XMLHttpRequest();
+
+ function resolve(xhttp) {
+
+    let response = JSON.parse(event.target.response);
+
+     document.getElementById("hidden").innerHTML = response.value.joke;
+
+     if (xhttp.readyState === XMLHttpRequest.DONE) {
+       if (xhttp.responseText.value) {
+         document.getElementById("hidden").innerHTML  = xhttp.responseText.value.joke;
+       }
+     }
+ }
+
+ function reject() {
+   //exercise 8
+   let obj = document.getElementById("hidden");
+   obj.style.color ="red";
+ }
+
+ eventRequestReusable(config).then(resolve,reject);
+}
+
+function configAjax (methodHttp, url, asyncronic){
+ this.methodHttp = methodHttp;
```

Comment:
take notice that this is adding a methodHttp variable to the global object, be careful with that, you can cause unexpected behaviour, instead change it to something like 
````
function configAjax (method, url, async) {
    var config = {
        method: method,
        url: url,
        asyncronic:  async
    };

    return config;
}
````

and the call it with:
``let httpConfig = configAjax('GET', 'http://myurl.com', true)``